### PR TITLE
Allow PublicSign=true even if full keys are available

### PR DESCRIPTION
**Pull request by omajid**
In certain environments - such as RHEL 9 - full signing does not work. That's because full signing requires SHA1 which is considered weak and was disabled in OpenSSL. Trying to use full signing leads to a Interop+Crypto+OpenSslCryptographicException. For more details, see https://github.com/dotnet/runtime/issues/65874.

In contrast, public signing doesn't use SHA1 and works fine in these environments.

To make sure we can still build projects in those environments using arcade, allow arcade consumers to select public signing even when we have all the keys for full signing.

Fixes: #12515

### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

**Comment by omajid** [comment]
cc @mmitche

**Comment by mmitche** [comment]
LGTM. Usually we'd put tests for this in arcade-validation (testing msbuild isn't always the easiest). IMO though, this particular issue is niche enough that if we will catch it in source build validation, it's fine to simply make a fix here.

**Review by mmitche — APPROVED** [review]

File: src/Microsoft.DotNet.Arcade.Sdk/tools/StrongName.targets [hunk 1]
```diff
@@ -5,6 +5,7 @@
   <!--
     Reads variables:
       SignAssembly    "true" to sign the output assembly of the current project
+      FullAssemblySigningSupported    "false" to use public signing even when full signing is possible
```
Comment (mmitche) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/StrongName.targets:8]: Can you put a comment here with some context on when `FullAssemblySigningSupported` might be set to false?

Comment (omajid) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/StrongName.targets]: Does the new comment read okay? Any suggestions on the details or phrasing?

File: src/Microsoft.DotNet.Arcade.Sdk/tools/StrongName.targets [hunk 2]
```diff
@@ -5,6 +5,12 @@
   <!--
     Reads variables:
       SignAssembly    "true" to sign the output assembly of the current project
+      FullAssemblySigningSupported    "false" to use public signing even when full signing is possible. This is useful
+                                      in environments where full signing is non-functional or not desired. For example,
+                                      in some Linux distributions RSA+SHA1 (required for full signing) is not
+                                      functional/availalbe, and trying to use full signing results in the runtime
```
Comment (mmitche) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/StrongName.targets:11]:
```suggestion
                                      functional/available, and trying to use full signing results in the runtime
```

**Review by mmitche — APPROVED** [review]
Thanks. These types comments are pure gold in the future.

**Review by mmitche — APPROVED** [review]

**Pull request merged**